application/src/main/java/org/togetherjava/tjbot/commands/filesharing/ShareLongAttachmentsMessageListener.java

```diff
@@ -0,0 +1,174 @@
+        try {
+            byte[] bytes = stream.readAllBytes();
+            return new String(bytes, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private HttpResponse<String> uploadToSharingService(MessageReceivedEvent event,
+            Message.Attachment attachment, String content) {
+        GistFile file = new GistFile();
+        file.setContent(content);
+
+        String parsedContent;
+        try {
+            parsedContent = JSON.writeValueAsString(file);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Couldn't parse content!", e);
+        }
+
+        String fileName = attachment.getFileName();
+        String fileExtension = attachment.getFileExtension();
+
+        if (fileExtension == null || fileExtension.equals("txt")) {
+            fileExtension = "java";
+        } else if (fileExtension.equals("fxml")) {
+            fileExtension = "xml";
+        }
+
+        fileName = fileName.substring(0, fileName.lastIndexOf(".")) + "." + fileExtension;
+
+        String body = """
+                {
+                  "description": "%s",
+                  "public": false,
+                  "files": {
+                      "%s": %s
+                  }
+                }""".formatted(event.getAuthor(), fileName, parsedContent);
+
+
+        HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create(API))
+            .header("Accept", "application/json")
+            .header("Authorization", "token " + API_KEY)
+            .POST(HttpRequest.BodyPublishers.ofString(body))
+            .build();
+
+        try {
+            return CLIENT.send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void sendMessage(MessageReceivedEvent event, Message.Attachment attachment,
+            HttpResponse<String> response) {
+        if (response.statusCode() < 200 || response.statusCode() > 299) {
+            logger.warn("Gist API unexpected response: " + response.body());
+            throw new IllegalStateException("Gist API unexpected response: " + response.body());
+        }
+
+        Message message = event.getMessage();
+        String fileName = attachment.getFileName();
+
+        GistResponse gistResponse = null;
+        try {
+            gistResponse = JSON.readValue(response.body(), GistResponse.class);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Couldn't parse response...", e);
+        }
+
+        String url = gistResponse.getHtmlUrl();
+
+        message.reply(
+                "I uploaded your file as gist, it is much easier to read for everyone like that especially for mobile users!")
```

Comment:
* missing comma `that, especially for`
* please remove the `!` that sounds unfriendly. maybe even add a thumbsup smiley (unicode)